File: connectors/connector-trello/src/main/scala/com/taskadapter/connector/trello/TrelloClient.scala
package com.taskadapter.connector.trello

import com.julienvey.trello.domain.Board
import com.taskadapter.http.HttpCaller

class TrelloClient(key: String, token: String) {
  val credentials = s"key=$key&token=$token"
  val baseUrl = s"https://api.trello.com/1"

  def createBoard(boardName: String): Board = {
    val createBoardUrl = baseUrl + s"/boards?name=${boardName}&${credentials}"
    val board = HttpCaller.post(createBoardUrl, classOf[Board])
    board
  }

  def closeBoard(boardId: String): Board = {
    val url = baseUrl + s"/boards/$boardId?closed=true&${credentials}"
    val board = HttpCaller.put(url, classOf[Board])
    board
  }

  def getBoards(memberIdOrLogin: String): Seq[Board] = {
    val url = baseUrl + s"/members/$memberIdOrLogin/boards?${credentials}"
    val result = HttpCaller.get(url, classOf[Array[Board]])
    result
  }

}

File: connectors/connector-trello/src/main/java/com/taskadapter/connector/trello/TrelloClient.java
package com.taskadapter.connector.trello;

import com.julienvey.trello.domain.Board;
import com.taskadapter.http.HttpCaller;

import java.io.IOException;
import java.util.List;

public class TrelloClient {
    private final String credentials;
    private final String baseUrl;

    public TrelloClient(String key, String token) {
        credentials = "key=" + key + "&token=" + token;
        baseUrl = "https://api.trello.com/1";
    }

    Board createBoard(String boardName) throws IOException {
        // TODO 14 name with spaces??
        var createBoardUrl = baseUrl + "/boards?name=" + boardName + "&" + credentials;
        var board = HttpCaller.post(createBoardUrl, Board.class);
        return board;
    }

    Board closeBoard(String boardId) throws IOException {
        var url = baseUrl + "/boards/" + boardId + "?closed=true&" + credentials;
        var board = HttpCaller.put(url, Board.class);
        return board;
    }

    public List<Board> getBoards(String memberIdOrLogin) {
        var url = baseUrl + "/members/" + memberIdOrLogin + "/boards?" + credentials;
        try {
            return HttpCaller.get(url, List.class);
        } catch (IOException e) {
            throw new RuntimeException(e.toString(), e);
        }
    }
}
